revise cipher suites section (fixes issue #53)

davegarrett opened the pull request:
This revises "The Cipher Suites" section to simplify redundant language, actually explain the de facto naming convention, and add all currently available AEAD cipher suites. It also drops the appendix table that has already had its section whittled down quite a bit, in favor of the explanation and new table in the main section.

Note that this is all of the AEAD ciphers available in the current IANA registry [0], plus ChaChaPoly (which has to be a normative reference [1], so must reach RFC status before this does). Issue #53 just said "standards track AEAD ciphers", but as the registry lists them all, adding them wasn't difficult. If we'd rather not cite them here then they can of course be cut out (ARIA, Camellia, & ECDHE AES-CCM are informational; the rest are standards track, including DHE AES-CCM). Alternatively, the table could be annotated to note which are on what track or split in two.

This has no changes to any requirements.

[0] https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-4
[1] https://www.ietf.org/iesg/statement/normative-informative.html

richsalz commented on draft-ietf-tls-tls13.md:
```diff
@@ -3056,9 +3063,29 @@ This section describes protocol types and constants.
+via hello messages in the TLS handshake.
+Cipher suites follow a general naming convention:
+
+       CipherSuite TLS_KEA_SIGN_WITH_CIPHER_HASH = VALUE;
```
perhaps use uppercase for TLS WITH and lowercase for kea sign cipher hash?

davegarrett replied:
I could explicitly mention that the "TLS" & "WITH" parts are just static parts of the naming. That could make it more clear, as well.

richsalz commented on draft-ietf-tls-tls13.md:
```diff
@@ -3066,52 +3093,69 @@ negotiated, as it provides no more protection than an unsecured connection.
+AES-GCM [RFC5288], ARIA-GCM [RFC6209], and CAMELLIA-GCM [RFC6367].
+
+All cipher suites in this section are specified for usage with TLS 1.3 as well
+as with TLS 1.2 in implementations acting in a backwards compatible mode.
```
TLS 1.2 in backward compat mode?  This sentence is confusing.

davegarrett commented:
@richsalz I addressed your comments in this commit. I switched to a table to make things easier to read and explicitly clarified the constant string bits so as to be more clear. I also dropped the confusing sentence fragment.

richsalz commented:
looks good.

davegarrett commented:
AES-OCB will also be addable to this list, but its codepoints are all completely TBD still.
https://tools.ietf.org/html/draft-zauner-tls-aes-ocb-02

davegarrett commented:
Rebased, squashed, & updated for recent changes to draft.

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -3413,52 +3450,73 @@ negotiated, as it provides no more protection than an unsecured connection.
+    TLS_DHE_RSA_WITH_CHACHA20_POLY1305            {0xCC,0xA3}  [I-D.ietf-tls-chacha20-poly1305]
+
+[[TODO: CHACHA20_POLY1305 cipher suite IDs are TBD. Must update above if
+final versions change.]]
```
My understanding is that they were definitely changing the numbers. If that's so, let's just put TBD here.

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -3413,52 +3450,73 @@ negotiated, as it provides no more protection than an unsecured connection.
+DHE AES GCM & CCM are standards track ([RFC5288] & [RFC6655]),
+as is ChaChaPoly ([I-D.ietf-tls-chacha20-poly1305]).  The ECDHE variants of AES
+and other ciphers above are currently informational, however ECDHE AES GCM
+is widely deployed.
```
Let's put this in a table instead.

davegarrett replied:
On Wednesday, July 15, 2015 02:22:12 pm ekr wrote:
> Let's put this in a table instead.

Which would you prefer:
a) add an asterisk next to the spec links to indicate informational
b) split the table into two for standard and informational

I think (a) is better, because that keeps ECDHE AES GCM next to DHE AES GCM. Either works.

Is it possible to promote cipher suite codepoints to standard status straight in this RFC?


Dave

ekr replied:
Let's split the table.

I think we can in fact promote them, but 4492bis will be making them standards track anyway.

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -3413,52 +3450,73 @@ negotiated, as it provides no more protection than an unsecured connection.
+
+All cipher suites in this section are specified for usage with both TLS 1.2
+and TLS 1.3. (see {{backward-compatibility}})
+
```
And DTLS.

davegarrett commented:
Ok, all comments addressed with that last commit.

I've split the table into standards and non-standards and simplified the note to just mention ECDHE AES GCM being widely deployed.

davegarrett commented:
Rebased.

davegarrett commented:
Ok, reworded that bit too.
